eventio/search_utils.py

```diff
@@ -34,3 +34,23 @@ def find_all_subobjects(f, structure, level=0):
         elif isinstance(o, elem):
             objects.extend(find_all_subobjects(o, structure, level + 1))
     return objects
+
+
+def yield_subobjects(f, eventio_type):
+    '''yield subobjects of type, regardless of structure'''
+    if isinstance(f, eventio_type):
+        yield f
+    else:
+        try:
```

Comment:
`If o.header.only_subobjects`

Reply:
I think you mean:
```python
    if isinstance(f, eventio_type):
        yield f
    else:
        if f.header.only_subobjects:
            for o in f:
                yield from yield_subobjects(o, eventio_type)
```
but this does not work, since `f` in the beginning is an `EventIOFile` and then one gets:
    AttributeError: 'EventIOFile' object has no attribute 'header'

Another explicit type check for `EventIOFile` would work but feels bad.
I think this is a hint, that `EventIoFile itself should maybe get a `header` :-D ...

Reply:
I am going to merge this for the time being .. we can later make this function nicer if you like. The interface will most likely not change.